src/main/java/org/dependencytrack/event/kafka/processor/IntegrityAnalysiResultProcessor.java

```diff
@@ -0,0 +1,115 @@
+import com.github.packageurl.PackageURL;
+import io.micrometer.core.instrument.Timer;
+import org.apache.kafka.streams.processor.api.Processor;
+import org.apache.kafka.streams.processor.api.Record;
+import org.dependencytrack.model.Component;
+import org.dependencytrack.model.IntegrityAnalysisComponent;
+import org.dependencytrack.persistence.QueryManager;
+import org.hyades.proto.repometaanalysis.v1.HashMatchStatus;
+import org.hyades.proto.repometaanalysis.v1.IntegrityResult;
+
+import javax.jdo.JDODataStoreException;
+import javax.jdo.PersistenceManager;
+import javax.jdo.Query;
+import javax.jdo.Transaction;
+import java.util.Date;
+import java.util.UUID;
+
+public class IntegrityAnalysiResultProcessor implements Processor<String, IntegrityResult, Void, Void> {
+    private static final Logger LOGGER = Logger.getLogger(IntegrityAnalysiResultProcessor.class);
+    private static final Timer TIMER = Timer.builder("repo_meta_result_processing")
+            .description("Time taken to process repository meta analysis results")
+            .register(Metrics.getRegistry());
+
+    @Override
+    public void process(Record<String, IntegrityResult> record) {
+        final Timer.Sample timerSample = Timer.start();
+        try (final var qm = new QueryManager()) {
+            synchronizeComponentIntegrity(qm.getPersistenceManager(), record);
+        } catch (Exception e) {
+            LOGGER.error("An unexpected error occurred while processing record %s".formatted(record), e);
+        } finally {
+            timerSample.stop(TIMER);
+        }
+    }
+
+    private void synchronizeComponentIntegrity(final PersistenceManager pm, final Record<String, IntegrityResult> record) {
+
+        final IntegrityResult result = record.value();
+
+        final PackageURL purl;
+        try {
+            purl = new PackageURL(result.getComponent().getPurl());
+        } catch (MalformedPackageURLException e) {
+            LOGGER.warn("""
+                    Received repository integrity information with invalid PURL,\s
+                    will not be able to correlate; Dropping
+                    """, e);
+            return;
+        }
+
+        final Transaction trx = pm.currentTransaction();
+        try {
+            trx.begin();
+            final Query<IntegrityAnalysisComponent> query = pm.newQuery(IntegrityAnalysisComponent.class);
+            query.setFilter("repositoryIdentifier == :repository && component.id == :id && component.uuid == :uuid");
+            query.setParameters(
+                    record.value().getRepository(),
+                    record.value().getComponent().getComponentId(),
+                    UUID.fromString(record.value().getComponent().getUuid())
+            );
+            IntegrityAnalysisComponent persistentIntegrityResult = query.executeUnique();
+            if (persistentIntegrityResult == null || persistentIntegrityResult.getComponent() == null) {
```

Comment:
component is not nullable in db so getComponent() should not be null. we can still keep the check though